instrumentation/jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JDBCConnectionUrlParser.java

```diff
@@ -159,7 +188,19 @@
         serverName = serverName.substring(0, instanceLoc);
       }
 
-      int portLoc = serverName.indexOf(":");
+      Matcher ipv6Matcher = IPv6.matcher(serverName);
+      boolean isIpv6 = ipv6Matcher.find();
+
+      int portLoc = -1;
+      if (isIpv6) {
+        if (serverName.startsWith("[")) {
+          portLoc = serverName.indexOf("]:") + 1;
+        } else {
+          serverName = "[" + serverName + "]";
```

Comment:
There cannot be port in this case?

Reply:
There isn't one, but because the `shortUrl` contains the port number, it needs to be formatted that way
`shortUrl` example `sqlserver://[2001:0db8:85a3:0000:0000:8a2e:0370:7334]:1433`. I'm now thinking though, whether this is not appropriate for the `host` attribute i.e. `[2001:0db8:85a3:0000:0000:8a2e:0370:7334]`. What do you think?

By the way, the `mariadb` example is partially incorrect because it includes the port as part of the `shortUrl`

```
"jdbc:mariadb:loadbalance://[2001:0660:7401:0200:0000:0000:0edf:bdd7]:33,mdb.host/mdbdb"|null|"mariadb:loadbalance://2001:0660:7401:0200:0000:0000:0edf:bdd7:33"|"mariadb"|"loadbalance"|null|"2001:0660:7401:0200:0000:0000:0edf:bdd7"|33|null|"mdbdb"
```

Reply:
You lost me :) I am not proficient with ip6 formats. I just see:
```
if (serverName.startsWith("[")) {
   portLoc = XXX
} else {
  serverName = "[" + serverName + "]";
}
```
and wonder, why if `serverName` does not start with `[`, then it means there is no port?

Reply:
Apologies, I went neck-deep into IPv6 land. Yes, since `:` is a delimiter for the IPv6 parts, the only way to define a port in a URL/URI-like manner, is to wrap the address in `[`-`]` and append the port with a `:`.

Does this make sense, especially with regards to my point about the MariaDB tests?

Reply:
Clarification, when I say 

>the only way to define a port in a URL/URI-like manner

, I should have said instead, "the only _unambigious_ way to define a port in a URL/URI-like manner"

Reply:
I see. I have left a comment in #1421, but that is a separate issue.